src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -208,108 +209,100 @@ namespace OpenMS
       }
     }
 
-    if (sort_by_position_) spectrum.sortByPosition(); // this takes about 40% of the time
-    return;
+    if (sort_by_position_) spectrum.sortByPositionPresorted(chunks);
   }
 
 
   void TheoreticalSpectrumGenerator::addAbundantImmoniumIons_(PeakSpectrum& spectrum, const AASequence& peptide, DataArrays::StringDataArray& ion_names, DataArrays::IntegerDataArray& charges) const
   {
     Peak1D p;
 
```

Comment:
add a comment here, that these mass need to stay sorted

Reply:
What do you mean by that?